txstratum/filters.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) Hathor Labs and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Any, Optional, Set
+
+from hathorlib.scripts import P2PKH
+from structlog import get_logger
+
+from txstratum.toi_client import TOIError
+
+if TYPE_CHECKING:
+    from hathorlib import BaseTransaction
+
+    from txstratum.toi_client import TOIAsyncClient
+
+
+logger = get_logger()
+
+
+class TXFilter(ABC):
+    """Base class for tx filters."""
+
+    @abstractmethod
+    async def check_tx(self, tx: 'BaseTransaction', data: Any) -> bool:
+        """Return if the tx should be blocked."""
+        raise NotImplementedError
+
+    @abstractmethod
+    async def close(self) -> None:
+        """Close any open session."""
+        raise NotImplementedError
+
+
+class FileFilter(TXFilter):
+    """Filter tx based on a set of banned tx_ids and addresses."""
+
+    def __init__(self, banned_tx_ids: Set[bytes] = set(), banned_addresses: Set[str] = set()) -> None:
```

Comment:
Maybe create a new issue to make `FileFilter` check for the file last modified date and automatically update in case of a file update.